src/model.ts

```diff
@@ -1,4 +1,68 @@
 import WebSocket = require("ws");
+
+interface TimerBase {
+  isRunning: boolean;
+  timeLeft: number;
+}
+
+class Timer implements TimerBase {
+  isRunning: boolean;
+  #startTime: number;
+  timeLeft: number;
+  #intervalID;
+
+  constructor(allotedTime: number) {
+    this.isRunning = false;
+    this.#startTime = 0;
+    this.timeLeft = allotedTime;
+  }
+
+  #getTimeElapsedSinceLastStart() {
+    if (!this.#startTime) {
+      return 0;
+    }
+
+    return Date.now() - this.#startTime;
+  }
+  start() {
+    if (this.isRunning) {
+      throw "Timer is already running";
+    }
+    this.isRunning = true;
+    this.#startTime = Date.now();
+
+    this.#intervalID = setInterval(() => {
+      this.timeLeft = this.timeLeft - this.#getTimeElapsedSinceLastStart();
+      this.#startTime = Date.now();
+      console.log(this.timeLeft);
+      if (this.timeLeft <= 0) {
+        this.stop();
+        throw "Player Timed Out";
+      }
+    }, 250);
+  }
+
+  stop() {
+    if (!this.isRunning) {
+      throw "Timer is already stopped";
+    }
+    this.timeLeft = this.timeLeft - this.#getTimeElapsedSinceLastStart();
+    this.isRunning = false;
+    clearInterval(this.#intervalID);
+  }
+
+  reset() {
+    this.timeLeft = 0;
```

Comment:
We should reset the values of all variables here to what we do in the `constructor`. We can then call `reset` inside the `constructor`